tfe/resource_tfe_workspace_test.go

```diff
@@ -1326,3 +1360,25 @@ resource "tfe_workspace" "foobar" {
 }
 `, rInt)
 }
+
+func testAccTFEWorkspace_remoteState(rInt int) string {
+	return fmt.Sprintf(`
+resource "tfe_organization" "foobar" {
+  name  = "tst-terraform-%d"
+  email = "admin@company.com"
+}
+resource "tfe_workspace" "foobar_remote" {
+  name               = "workspace-test-2"
+  organization       = tfe_organization.foobar.id
+  global_remote_state = false
+}
+
+resource "tfe_workspace" "foobar" {
+  name               = "workspace-test"
+  organization       = tfe_organization.foobar.id
+  allow_destroy_plan = false
+  auto_apply         = true
+	global_remote_state = false
+  remote_state_consumer_ids = [tfe_workspace.foobar_remote.id]
+}`, rInt)
+}
```

Comment:
We should add a slightly more thorough test, perhaps moving from global sharing to two different configs of different state consumers (testing adding/removing them in the Update function).

Reply:
My bad, i intended to add more tests and then didn't. Will do so

Reply:
added a few more